pkgs/by-name/ca/caddy/package.nix

```diff
@@ -61,11 +64,58 @@ buildGoModule {
       --zsh <($out/bin/caddy completion zsh)
   '';
 
-  passthru.tests = {
-    inherit (nixosTests) caddy;
-    version = testers.testVersion {
-      command = "${caddy}/bin/caddy version";
-      package = caddy;
+  passthru = {
+    withPlugins =
+      {
+        caddyModules,
+        vendorHash ? lib.fakeHash,
+      }:
+      buildGoModule {
+        pname = "${caddy.pname}-with-plugins";
+        inherit (caddy)
```

Comment:
One concern, this might result in confusing behavior with overrides, as `(caddy.override ...).withPlugins ...` would not work as expected. This could be resolved by inheriting from the outer scope instead of `(caddy)` I believe.

Reply:
Thank you for the suggestion! I [restructured](https://github.com/NixOS/nixpkgs/pull/317881/commits/db5046f7caef06498835c47185a9035f80b446b3) the code to change its scoping so that `caddy.withPlugins` inherits from the outer scope. For some reason, though, this doesn't seem to fix `overrideAttrs` done on `caddy` when `caddy.withPlugins` is used. Would you suggest I add documentation that recommends using `overrideAttrs` on `caddy.withPlugins` instead of on `caddy` when deriving `caddy-with-plugins`?

Reply:
Getting it to work correctly with `overrideAttrs` will probably require something like the `finalAttrs` pattern for `buildGoModule`, which we don't have yet. Unsure on the docs mention.